### DISCUSSION TITLE
Play video using Low-Latency HLS stream in browser failed

### DISCUSSION THREAD

**Discussion by zt1983811** [bluenviron/mediamtx#1166] [Questions]
# Context:
We would like to use the recent Low-Latency HLS protocol for streaming video in browser to replace using webrtc

# Problem:
We followed the instruction in [Readme HLS section](https://github.com/aler9/rtsp-simple-server#hls-protocol). However I tried in different browser (chrome, safari) with <ifroame> <video> and hls.js (react). It seems able to get the stream form network but the video never start playing. Please see the screenshot attachment below.

![Untitled_ Sep 22, 2022 11_05 PM](https://user-images.githubusercontent.com/2559543/191885184-b02af997-b850-4f0c-a592-39206d2be2ff.gif)

# Steps to produce the problem:
1. please check my local branch [config changes + certificate + 2 videos file I tested with ](https://github.com/aler9/rtsp-simple-server/compare/main...zt1983811:rtsp-simple-server:try/hls)
2. Just run the rtsp server by build and run in Golang (1.17)
3. then publish video stream to server by ffmpeg, I tried few different config for both video
```
ffmpeg -re -stream_loop -1 -i drone.mp4 -c copy -f rtsp -rtsp_transport tcp rtsp://localhost:8554/mystream
ffmpeg -re -stream_loop 1 -i video.mkv -pix_fmt yuv420p -c:v libx264 -preset ultrafast -b:v 600k -c:a aac -b:a 160k -f rtsp rtsp://localhost:8554/mystream
```
4. Try to load the url from browser `https://localhost:8888/mystream/`
in chrome:
![image](https://user-images.githubusercontent.com/2559543/191885799-f3f0c362-efca-4361-b326-fee57684b497.png)
in safari:
![image](https://user-images.githubusercontent.com/2559543/191885887-e7abac55-3f9a-4f0e-bcb8-1ad766909c32.png)

I used iframe video
```
<html>
    <body>
        <h1>Vidoe</h1>
<video src="http://localhost:8888/mystream/index.m3u8"></video>
<iframe src="https://localhost:8888/mystream" scrolling="no"></iframe>
</body>
</html>
````
![image](https://user-images.githubusercontent.com/2559543/191885907-b432cec6-1816-47c8-b538-3bfb89a60e7e.png)

I also tried hls.js
```
import "./App.css";
import ReactHlsPlayer from "react-hls-player/dist";

function App() {
  return (
    <div className="App">
      <ReactHlsPlayer
        id="live-video-id" class="video-box-recording-complete" 
        src="https://localhost:8888/mystream/index.m3u8"
        autoPlay={false}
        controls={true}
        width="100%"
      />
    </div>
  );
}

export default App;

```
![image](https://user-images.githubusercontent.com/2559543/191886053-bd80a5ac-7e8b-4c09-8ce5-4a41d0d6f485.png)

# Question:
I feel it could be the way I publish the video is wrong?
or maybe the video itself is not good because you mentioned HLS only supports a single H264 video track and a single AAC audio track ? You can check the video I used in my folked branch

Thanks so much for help

**Comment by aler9** [comment]
Hello, i tried replicating the issue with the latest server version (v0.21.1) and Safari on macOS, but in my case it worked perfectly.

Make sure that `stream_loop` is set to -1:

```
ffmpeg -re -stream_loop -1 -i video.mkv -pix_fmt yuv420p -c:v libx264 -preset ultrafast -b:v 600k -c:a aac -b:a 160k -f rtsp rtsp://localhost:8554/mystream
```

Then make sure that macOS and rtsp-simple-server are both up to date.

![Untitled](https://user-images.githubusercontent.com/46489434/213476510-100cf76b-87eb-4d40-acd4-e165950a6c76.jpg)

**Comment by bmquynhlinh** [comment]
Hi, I had the same problem and just found out that the src is not index.m3u8 but stream.m3u8
```
      <ReactHlsPlayer
        id="live-video-id" class="video-box-recording-complete" 
        src="https://localhost:8888/mystream/stream.m3u8"
        autoPlay={false}
        controls={true}
        width="100%"
      />
```

[a comment by an automated account is not shown]
